### ISSUE TITLE
Allow multiple layers to be submitted for each eye

### ISSUE THREAD

**Issue by LibreVR** [ValveSoftware/openvr#255]
The Oculus SDK allows you to submit multiple textures of different sizes for each eye which are composited on top of each other before being displayed. It would be convenient if the VRCompositor allowed you to submit more than one texture per eye and have those automatically composited.

Note that this is not the same as VROverlays, since it is not a texture displayed at a certain position. For now I will handle this type of composition myself.

**Comment by aleiby** [comment]
You can specify a tracked device parent index for overlays.  The headset's index is 0, so that will create a quad that is locked to the head and should get you most of the way there.

**Comment by LibreVR** [comment]
I did try that, but it was difficult getting a quad that did not get clipped by the near plane but still covers the screen pixel-perfect. Also, unlike Submit() I can only use a Side-by-Side stereo texture, though compositing that is a lot easier since I don't need to use shaders to scale the texture.

Assuming the chaperone uses the projection matrix reported by OpenVR I'll try and derive the correct size for the quad at the near plane. I'll keep this issue updated with my progress.

**Comment by LibreVR** [comment]
Okay, I tried it again, but I was unsuccessful, these are my findings:

1. Even though I can parent the stereoscopic overlay to the headset it seems it is drawn on a single quad in the world space. That means that due to the IPD distance between the camera the quad is offset and not centered in the eye. Since it is just a single quad I can't align it in one eye, because it would be further offset in the other one.

![headset-overlay](https://cloud.githubusercontent.com/assets/18057864/20140734/bd1e34be-a684-11e6-91e4-86eaa6e11644.png)

2. The Oculus SDK uses a pre-multiplied alpha and it doesn't seem that OpenVR expects that, causing blending errors.

**Comment by mtschoen** [comment]
Hey there! Did anyone make any progress on this? I would also like to use a multi-layer setup, but don't see a clear path for doing so.

**Comment by echuber2** [comment]
Why do you not render the scene back to front to frame buffers?

**Comment by mtschoen** [comment]
My case is pretty specific. I am injecting into an existing app and wanting to render a transparent layer on top of the app's graphics. I could write to the texture being submitted, or try the overlay situation described above, but I wonder which makes more sense.

**Comment by echuber2** [comment]
I guess I can't be of any help... if you can inject at the level of OpenGL API calls you could take the textures being submitted and render more framebuffers while sampling the previous textures, then resubmit that. I don't know the specifics of what you're doing though. (Sounds a little bit sketchy that you need to use code injection?)

**Comment by mtschoen** [comment]
Yeah we can actually get the textures being submitted and write to them, but our current code path just submits a new layer via Oculus layers so I'm hoping to stay as similar to that as possible.
